src/main/scala/io/iohk/ethereum/jsonrpc/PersonalService.scala

```diff
@@ -114,32 +119,47 @@ class PersonalService(
     }.getOrElse(Left(InvalidParams("unable to recover address")))
   }
 
-  def sendTransaction(request: SendTransactionWithPassphraseRequest): ServiceResponse[SendTransactionWithPassphraseResponse] = Future {
-    keyStore.unlockAccount(request.tx.from, request.passphrase).left.map(handleError).map { wallet =>
-      val hash = sendTransaction(request.tx, wallet)
-      SendTransactionWithPassphraseResponse(hash)
+  def sendTransaction(request: SendTransactionWithPassphraseRequest): ServiceResponse[SendTransactionWithPassphraseResponse] = {
+    val maybeWalletUnlocked = Future { keyStore.unlockAccount(request.tx.from, request.passphrase).left.map(handleError) }
+    maybeWalletUnlocked.flatMap {
+      case Right(wallet) =>
+        val futureTxHash = sendTransaction(request.tx, wallet)
+        futureTxHash.map(txHash => Right(SendTransactionWithPassphraseResponse(txHash)))
+      case Left(err) => Future.successful(Left(err))
     }
   }
 
 
-  def sendTransaction(request: SendTransactionRequest): ServiceResponse[SendTransactionResponse] = Future {
+  def sendTransaction(request: SendTransactionRequest): ServiceResponse[SendTransactionResponse] = {
     unlockedWallets.get(request.tx.from) match {
       case Some(wallet) =>
-        Right(SendTransactionResponse(sendTransaction(request.tx, wallet)))
+        val futureTxHash = sendTransaction(request.tx, wallet)
+        futureTxHash.map(txHash => Right(SendTransactionResponse(txHash)))
 
       case None =>
-        Left(AccountLocked)
+        Future.successful(Left(AccountLocked))
     }
   }
 
-  private def sendTransaction(request: TransactionRequest, wallet: Wallet): ByteString = {
-    val defaultNonce = getCurrentAccount(request.from).getOrElse(Account.Empty).nonce
-    val tx = request.toTransaction(defaultNonce)
-    val stx = wallet.signTx(tx)
+  private def sendTransaction(request: TransactionRequest, wallet: Wallet): Future[ByteString] = {
+    implicit val timeout = Timeout(miningConfig.poolingServicesTimeout)
```

Comment:
I'm wondering whether `poolingServicesTimeout` should be moved somewhere else. Having a TX pool doesn't have to mean we're doing any mining. `PersonalService` is also not dependent on mining, so passing the mining config to feels kinda strange.